bcftbx/cmdparse.py

```diff
@@ -234,90 +242,104 @@ def print_command(self,cmd,message=None):
 # Functions
 #######################################################################
 
-def add_nprocessors_option(parser,default_nprocessors,default_display=None):
+def add_nprocessors_option(parser, default_nprocessors, default_display=None):
     """Add a '--nprocessors' option to a parser
-    
-    Given an OptionParser 'parser', add a '--nprocessors' option.
+
+    Given an ArgumentParser 'parser', add a '--nprocessors' option.
 
     The value of this option can be accessed via the 'nprocessors'
     attribute of the parser options.
 
     If 'default_display' is not None then this value will be shown
     in the help text, rather than the value supplied for the default.
 
-    Returns the input OptionParser object.
+    Returns the input ArgumentParser object.
 
     """
     if default_display is None:
         default_display = default_nprocessors
-    parser.add_option('--nprocessors',action='store',
-                      dest='nprocessors',default=default_nprocessors,
-                      help="explicitly specify number of processors/cores to use "
-                      "(default %s)" % 
-                      default_display)
+    if isinstance(parser, argparse.ArgumentParser):
+        add_cmd = parser.add_argument
+    else: add_cmd = parser.add_option
+    add_cmd('--nprocessors', action='store',
+            dest='nprocessors', default=default_nprocessors,
+            help="explicitly specify number of processors/cores to use "
+            "(default %s)" %default_display)
     return parser
 
 def add_runner_option(parser):
     """Add a '--runner' option to a parser
-    
-    Given an OptionParser 'parser', add a '--runner' option.
+
+    Given an ArgumentParser 'parser', add a '--runner' option.
 
     The value of this option can be accessed via the 'runner'
     attribute of the parser options (use the 'fetch_runner'
     function to return a JobRunner object from the supplied
     value).
 
-    Returns the input OptionParser object.
+    Returns the input ArgumentParser object.
 
     """
-    parser.add_option('--runner',action='store',
-                      dest='runner',default=None,
-                      help="explicitly specify runner definition (e.g. "
-                      "'GEJobRunner(-j y)')")
+    if isinstance(parser, argparse.ArgumentParser):
+        add_cmd = parser.add_argument
+    else: add_cmd = parser.add_option
+    add_cmd('--runner', action='store',
```

Comment:
It's preferable to split the clause following the `else` statement onto a separate new line.